plugins/rikki/heroeslounge/classes/discord/Attendance.php

```diff
@@ -68,8 +67,82 @@ public function CheckAttendance()
 
     public function IsOnServer($discordTag)
     {
-        $presentDiscordTags = $this->FetchUsers();
-        
-        return ($presentDiscordTags != false) ? in_array($discordTag, $presentDiscordTags) : false;
+        $presentUsers = $this->FetchUsers();
+        $userId = '';
+
+        if ($presentUsers) {
+          foreach ($presentUsers as $user) {
+            $userTag = $user["username"] . "#" . $user["discriminator"];
+            if ($userTag == $discordTag) {
+              $userId = $user["id"];
+              break;
+            }
+          }
+        }
+        return $userId
+    }
+
+    public function CheckIndividualAttendance($discordId)
+    {
+      $url = 'https://discordapp.com/api/guilds/200267155479068672/members/';
+
+      $auth_header = AuthCode::getCode();
+      $headers = [
+        "Content-Type:application/x-www-form-urlencoded",
+        $auth_header,
+        "User-Agent: HeroesLounge (http://heroeslounge.gg, 0.1)"
+      ];
+
+      $ch = curl_init($url . $discordId);
+      curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+      curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+      $output = curl_exec($ch);
+
+      if (curl_errno($ch)) {
+          return false;
+      }
+
+      $memberData = json_decode($output, true);
+
+      curl_close($ch);
+
+      return (isset($memberData)) ? true : false;
+
+    }
+
+    public function CreateDiscordTagArray($users)
+    {
+      $presntDiscordTags = [];
+
+      foreach ($users as $user) {
+        $presentDiscordTags[] = $user["username"] . "#" . $user["discriminator"];
+      }
+
+      return $presentDiscordTags;
+    }
+
+    /*
+      Unsure if this is being done correctly and if it should not instead be inside of the migration file?
+    */
+    public function migrationDiscordTagsToIds ()
+    {
+      $sloths = Db::table('rikki_heroeslounge_sloths')->select('id', 'discord_tag');
```

Comment:
Again, style. Do Sloth::all() and access individual fields with $sloth->id and $sloth->discord_tag

Reply:
Made some changes, unsure if 100% correct now.